### PR TITLE
Release/0.7.0

### PR DESCRIPTION
**Pull request by ryan-summers**
This PR reorganizes the README a little bit and adds a `dsl.md` file that is included in both the lib.rs file and the README (as a reference) so that the syntax is easily accessible on docs.rs

This also prepares for a 0.7.0 release

### CODE REVIEW

**Comment by dkumsh** [comment]
@ryan-summers 
I wonder if you could find some time to kick of this release with the most recent changes.

File: CHANGELOG.md [hunk 1]
```diff
@@ -172,7 +174,8 @@ a long list of states to go through.
 * Support for data in states
 * Change log added
 
-[Unreleased]: https://github.com/korken89/smlang-rs/compare/v0.6.0...master
+[Unreleased]: https://github.com/korken89/smlang-rs/compare/v0.7.0...master
+[v0.6.0]: https://github.com/korken89/smlang-rs/compare/v0.6.0...v0.7.0
```
Comment (korken89) [inline CHANGELOG.md:178]: Copy paste error

File: Cargo.toml [hunk 2]
```diff
@@ -11,7 +11,7 @@ edition = "2018"
 readme = "README.md"
 
 [dependencies]
-smlang-macros = { path = "macros", version = "0.6.0" }
+smlang-macros = { path = "macros", version = "0.7.0" }
 async-trait = "0.1"
```
Comment (korken89) [inline Cargo.toml:16]: Is async trait a required dependency?

File: README.md [hunk 3]
```diff
@@ -315,6 +360,7 @@ List of contributors in alphabetical order:
 
 * Emil Fresk ([@korken89](https://github.com/korken89))
 * Mathias Koch ([@MathiasKoch](https://github.com/MathiasKoch))
+* Ryan Summers ([@ryan-summers](https://github.com/ryan-summers))
```
Comment (korken89) [inline README.md:363]: Perfect that you added yourself!

**Review by korken89 — CHANGES_REQUESTED** [review]
Found one error and have one question

File: CHANGELOG.md (hunk 1 above)
Comment (ryan-summers) [reply CHANGELOG.md]: Fixed in 9b97da80106831cf6d35edc7a9e53eeb04ebea7c, thanks for the catch

**Comment by ryan-summers** [comment]
I also seem to have nearly forgotten to update Cargo.toml with 0.7.0 - fixed in 610ce45d416abb6c91c88606e8b47cbaaf595925

**Review by korken89 — APPROVED** [review]
Looks good, thank you!

**Pull request merged**